src/shared/components/query/QueryStore.ts

```diff
@@ -473,20 +417,46 @@ export class QueryStore
 		default: {},
 	});
 
-	@computed get selectableStudiesSet():{[studyId:string]:boolean} {
-		const ret = Object.assign({}, this.cancerStudyIdsSet.result);
-		for (const cohort of this.virtualCohorts) {
-			ret[cohort.id] = true;
-		}
-		return ret;
-	}
+	readonly virtualStudies = remoteData(sessionServiceClient.getUserVirtualStudies(), []);
+
+	readonly selectableIdToPhysicalIdsMap = remoteData<{[studyId:string]:string[]}>({
+		await: ()=>[this.cancerStudies, this.virtualStudies],
+	    invoke: async ()=>{
+			let physicalStudiesIdsMap:{[studyId:string]:string[]} = this.cancerStudies.result.reduce((obj:{[studyId:string]:string[]}, item) =>{
+				obj[item.studyId] = [item.studyId]
+				return obj
+			}, {});
+
+			let virtualStudiesIdsMap:{[studyId:string]:string[]} = this.virtualStudies.result.reduce((obj:{[studyId:string]:string[]}, item) =>{
+				obj[item.id] = [item.id]
+				return obj
+			}, {});
+
+			let knownSelectableIds:{[studyId:string]:string[]} = Object.assign({}, physicalStudiesIdsMap, virtualStudiesIdsMap);
+
+			const unknownIds:string[] = this._defaultSelectedIds.keys().filter(id => !knownSelectableIds[id]);
+
+			await Promise.all(unknownIds.map(id =>{
+				return new Promise((resolve, reject) => {
+					sessionServiceClient.getVirtualStudy(id).then((response)=>{
+						knownSelectableIds[id] = response.data.studies.map(study=>study.id)
+						resolve();
+					}).catch(() => {
+						resolve();

```

Comment:
I think we should handle this better.  If we get an error code if that id doesn't exists in session then we should message the user because the query is basically invalid.  also, we should wait until we have all responses before assigning to knownSelectableIds.    i believe that Promise.all().then() would give you aggregated collection of responses.  Make sense?

Reply:
I will let you know what the right way is to do an error message shortly.